packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -0,0 +1,269 @@
+  final Color color;
+  final double scale;
+  /// If this is true the image will be mirrored horizontally.
+  final bool shouldMirror;
+  final _UiPathFactory uiPathFactory;
+
+  @override
+  void paint(ui.Canvas canvas, Size size) {
+    canvas.scale(scale, scale);
+    if (shouldMirror) {
+      canvas.rotate(math.pi);
+      canvas.translate(-size.width, -size.height);
+    }
+
+    for (_PathFrames path in paths)
+      path.paint(canvas, color, uiPathFactory, progress.value.clamp(0.0, 1.0));
+  }
+
+
+  @override
+  bool shouldRepaint(_AnimatedIconPainter oldDelegate) {
+    return oldDelegate.progress.value != progress.value
+      || oldDelegate.color != color
+      // We are comparing the paths list by reference, assuming the list is
+      // treated as immutable to be more efficient.
+      || oldDelegate.paths != paths
+      || oldDelegate.scale != scale
+      || oldDelegate.uiPathFactory != uiPathFactory;
+  }
+
+  @override
+  bool hitTest(Offset position) => null;
+
+  @override
+  bool shouldRebuildSemantics(CustomPainter oldDelegate) => false;
+
+  @override
+  SemanticsBuilderCallback get semanticsBuilder => null;
+}
+
+class _PathFrames {
+  const _PathFrames({
+    @required this.commands,
+    @required this.opacities
+  });
+
+  final List<_PathCommand> commands;
+  final List<double> opacities;
+
+  void paint(ui.Canvas canvas, Color color, _UiPathFactory uiPathFactory, double progress) {
+    final double opacity = _interpolate(opacities, progress, lerpDouble);
+    final ui.Paint paint = new ui.Paint()
+      ..style = PaintingStyle.fill
+      ..color = color.withOpacity(opacity);
+    final ui.Path path = uiPathFactory();
+    for (_PathCommand command in commands)
+      command.apply(path, progress);
+    canvas.drawPath(path, paint);
+  }
+}
+
+abstract class _PathCommand {
+  const _PathCommand();
+
+  void apply(ui.Path path, double progress);
+}
+
+class _PathMoveTo extends _PathCommand {
+  const _PathMoveTo(this.points);
+
+  final List<Offset> points;
+
+  @override
+  void apply(Path path, double progress) {
+    final Offset offset = _interpolate(points, progress, Offset.lerp);
+    path.moveTo(offset.dx, offset.dy);
+  }
+}
+
+class _PathCubicTo extends _PathCommand {
+  const _PathCubicTo(this.controlPoints1, this.controlPoints2, this.targetPoints);
+
+  final List<Offset> controlPoints2;
+  final List<Offset> controlPoints1;
+  final List<Offset> targetPoints;
+
+  @override
+  void apply(Path path, double progress) {
+    final Offset controlPoint1 = _interpolate(controlPoints1, progress, Offset.lerp);
+    final Offset controlPoint2 = _interpolate(controlPoints2, progress, Offset.lerp);
+    final Offset targetPoint = _interpolate(targetPoints, progress, Offset.lerp);
+    path.cubicTo(
+      controlPoint1.dx, controlPoint1.dy,
+      controlPoint2.dx, controlPoint2.dy,
+      targetPoint.dx, targetPoint.dy
+    );
+  }
+}
+
+// ignore: unused_element
+class _PathLineTo extends _PathCommand {
```

Comment:
presumably the ignore can be removed now?

Reply:
not yet, current submitted assets don't use it

Reply:
Use it in a test?

Reply:
It's actually being used in a test, but due to the part of trick the analyzer still complains (https://github.com/amirh/flutter/blob/98c3a23deb7cde8893accb844543d65f2c1ff43f/packages/flutter/test/material_animated_icons/animated_icons_private_test.dart#L337)

Reply:
d'oh